client/ayon_core/plugins/publish/validate_asset_docs.py

```diff
@@ -37,3 +49,142 @@ def process(self, instance):
+            entity_data, data = entity_to_inspect.pop(0)
+
+            for name, value in data.items():
+                entity = entity_data.get(name)
+                child = value.get("children", None)
+
+                if entity and child:
+                    entity_children = {
+                        child.name: child
+                        for child in entity.children
+                    }
+                    entity_to_inspect.append((entity_children, child))
+
+                # Destination shot already exists return for validation.
+                elif (
+                    value.get("folder_type") == "Shot"
+                    and entity and not child
+                ):
+                    shot_data = value.get("attributes", {})
+                    return entity_hub, entity, shot_data
+
+        return None
+
+    def process(self, instance):
+        """ Validate existing shot resolution.
+        """
+        try:
+            hierarchy_context = instance.context.data["hierarchyContext"]
+
+        except KeyError:
+            self.log.info("No hierarchy context defined for shot instance.")
+            return
+
+        validation_data = self.get_shot_data(hierarchy_context)
+        if not validation_data:
+            self.log.info(
+                "Destination shot does not exist yet, "
+                "nothing to validate."
+            )
+            return
+
+        values = self.get_attr_values_from_data(instance.data)
+        _, entity, shot_data = validation_data
+
+        # Validate existing shot resolution is matching new one
+        # ask for confirmation instead of blind update, this prevents mistakes.
+        if values.get("updateExistingFolderResolution", True):
+            for resolution_attrib in _RESOLUTION_ATTRIBS:
+                shot_value = shot_data.get(resolution_attrib)
+                entity_value = entity.attribs.get(resolution_attrib)
+                if shot_value and shot_value != entity_value:
+                    raise PublishValidationError(
+                        "Resolution mismatch for shot."
+                        f"{resolution_attrib}={shot_value} but "
+                        f"already existing shot is set to {entity_value}."
+                    )
+
+        # If update existing shot is disabled, remove any resolution attribs.
+        else:
+            for resolution_attrib in _RESOLUTION_ATTRIBS:
+                shot_data.pop(resolution_attrib, None)
+
+            self.log.info(
+                "Ignore existing shot resolution validation "
+                "(update is disabled)."
+            )
+
+    @classmethod
+    def get_attr_defs_for_instance(
+        cls, create_context, instance,
+    ):
+        if instance.product_type not in cls.families:
+            return []
+
+        return [
+            BoolDef(
+                "updateExistingFolderResolution",
+                default=True,
+                label="Update existing shot resolution.",
```

Comment:
Avoid the dot on the label to match other attributes in the UI
```suggestion
                label="Update existing shot resolution",
```